File: backend/src/main/java/com/example/demo/model/Price.java
package com.example.demo.model;


import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;




@Entity
@Table(name = "price")
@IdClass(CompositeKey.class)
public class Price implements Serializable {

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "product_id", nullable = false)
    @JsonIgnore
    @Id
    private Product product;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "shop_id", nullable = false)
    @JsonIgnore
    @Id
    private Shop shop;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "volunteer_id", nullable = false)
    @JsonIgnore
    @Id
    private Volunteer volunteer;

    private Date timestamp;
    private Double value;

    public Product getProduct() {
        return product;
    }

    public void setProduct(Product product) {
        this.product = product;
    }

    public Shop getShop() {
        return shop;
    }

    public void setShop(Shop shop) {
        this.shop = shop;
    }

    public Volunteer getVolunteer() {
        return volunteer;
    }

    public void setVolunteer(Volunteer volunteer) {
        this.volunteer = volunteer;
    }

    public Date getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(Date timestamp) {
        this.timestamp = timestamp;
    }

    public Double getValue() {
        return value;
    }

    public void setValue(Double value) {
        this.value = value;
    }
}

File: app/models/price.ts
import { Product } from "./product";
import { User } from "./user";
import { Shop } from "./shop";

export class Price {
    _id: number;
    product: Product;
    user: User;
    shop: Shop;
    value: number;
    timestamp: Date;

    constructor(options: {
        _id?: number;
        product: Product;
        user: User;
        shop: Shop;
        value: number;
        timestamp: Date;
      }) {
        if ("_id" in options)
          this._id = options._id;
          this.product = options.product;
          this.user = options.user;
          this.shop = options.shop;
          this.value = options.value;
          this.timestamp = options.timestamp;
      }

      toJSON() {
        return Object.assign({}, this);
      }
}
